tests/abi/testabi_ks.c

```diff
@@ -1147,9 +1147,9 @@ static void test_ksLookupName ()
 	ksAppendKey (ks, keyNew ("user/named/key", KEY_VALUE, "myvalue", KEY_END));
 	ksAppendKey (ks, keyNew ("system/named/syskey", KEY_VALUE, "syskey", KEY_END));
 	ksAppendKey (ks, keyNew ("system/sysonly/key", KEY_VALUE, "sysonlykey", KEY_END));
-	ksAppendKey (ks, keyNew ("user/named/bin", KEY_BINARY, KEY_SIZE, 10, KEY_VALUE, "binary\1\2data", KEY_END));
-	ksAppendKey (ks, keyNew ("system/named/bin", KEY_BINARY, KEY_SIZE, 10, KEY_VALUE, "sys\1bin\2", KEY_END));
-	ksAppendKey (ks, keyNew ("system/named/key", KEY_BINARY, KEY_SIZE, 10, KEY_VALUE, "syskey", KEY_END));
+	ksAppendKey (ks, keyNew ("user/named/bin", KEY_BINARY, KEY_SIZE, sizeof ("binary\1\2data"), KEY_VALUE, "binary\1\2data", KEY_END));
```

Comment:
Ok, changing the library itself will not work. We can try to exclude the problematic pieces of the test with the `__SANITIZE_ADDRESS__` macro, but as the problematic keys are used several times after their declaration, this will be quite complex.

A last suggestion would be to fix the key sizes of above keys and introduce only one new key with a size larger than the value, which is then excluded for ASAN builds.